cpp/src/arrow/parquet/parquet-io-test.cc

```diff
@@ -120,90 +132,192 @@ class TestParquetIO : public ::testing::Test {
     return ReaderFromSink();
   }
 
- private:
+  std::unique_ptr<ParquetFileReader> TestFile(std::vector<T>& values, int num_chunks) {
+    std::shared_ptr<GroupNode> schema =
+        MakeSchema(test_traits<TestType>::parquet_enum, Repetition::REQUIRED);
+    std::unique_ptr<ParquetFileWriter> file_writer = MakeWriter(schema);
+    size_t chunk_size = values.size() / num_chunks;
+    for (int i = 0; i < num_chunks; i++) {
+      auto row_group_writer = file_writer->AppendRowGroup(chunk_size);
+      auto column_writer = static_cast<::parquet::TypedColumnWriter<
+          ::parquet::DataType<test_traits<TestType>::parquet_enum>>*>(
```

Comment:
You could simplify this (or, at least, make it reusable) with something like (have not tested, but for future reference)

```c++
template <typename T>
using ParquetType = ::parquet::DataType<test_traits<T>::parquet_enum>

template <typename T>
using ParquetWriter = ::parquet::TypedColumnWriter<ParquetType<T>>
```